ksql-engine/src/test/java/io/confluent/ksql/integration/UdfIntTest.java

```diff
@@ -2,317 +2,268 @@
-        TestUtils.tempDirectory(),
-        getClass().getClassLoader(),
-        value -> true, new UdfCompiler(Optional.empty()), Optional.empty(), true)
-        .load();
-
-    /**
-     * Setup test data
-     */
-    orderDataProvider = new OrderDataProvider();
-    itemDataProvider = new ItemDataProvider();
-    jsonRecordMetadataMap = testHarness.publishTestData(jsonTopicName,
-                                                        orderDataProvider,
-                                                        null,
-                                                        DataSource.DataSourceSerDe.JSON);
-    avroRecordMetadataMap = testHarness.publishTestData(avroTopicName,
-                                                        orderDataProvider,
-                                                        null,
-                                                        DataSource.DataSourceSerDe.AVRO);
-    testHarness.publishTestData(delimitedTopicName,
-                                itemDataProvider,
-                                null,
-                                DataSource.DataSourceSerDe.DELIMITED);
-    createOrdersStream();
+    final OrderDataProvider orderDataProvider = new OrderDataProvider();
+    final ItemDataProvider itemDataProvider = new ItemDataProvider();
 
-  }
+    jsonRecordMetadataMap = ImmutableMap.copyOf(TEST_HARNESS.publishTestData(JSON_TOPIC_NAME,
+        orderDataProvider,
+        null,
+        DataSource.DataSourceSerDe.JSON));
 
-  @After
-  public void after() throws Exception {
-    ksqlContext.close();
-    testHarness.stop();
-  }
+    avroRecordMetadataMap = ImmutableMap.copyOf(TEST_HARNESS.publishTestData(AVRO_TOPIC_NAME,
+        orderDataProvider,
+        null,
+        DataSource.DataSourceSerDe.AVRO));
 
-  @Test
-  public void testApplyUdfsToColumnsJson() throws Exception {
-    testApplyUdfsToColumns("SelectUDFsStreamJson".toUpperCase(),
-                           jsonStreamName,
-                           DataSource.DataSourceSerDe.JSON);
+    TEST_HARNESS.publishTestData(DELIMITED_TOPIC_NAME,
+        itemDataProvider,
+        null,
+        DataSource.DataSourceSerDe.DELIMITED);
   }
 
-  @Test
-  public void testApplyUdfsToColumnsAvro() throws Exception {
-    testApplyUdfsToColumns("SelectUDFsStreamAvro".toUpperCase(),
-                           avroStreamName,
-                           DataSource.DataSourceSerDe.AVRO);
+  @AfterClass
+  public static void classTearDown() {
+    TEST_HARNESS.stop();
   }
 
-  @Test
-  public void testShouldCastSelectedColumnsJson() throws Exception {
-    testShouldCastSelectedColumns("CastExpressionStream".toUpperCase(),
-                                  jsonStreamName,
-                                  DataSource.DataSourceSerDe.JSON);
+  public UdfIntTest(final DataSource.DataSourceSerDe format) {
+    this.format = format;
+    switch (format) {
+      case AVRO:
+        this.sourceTopicName = AVRO_TOPIC_NAME;
```

Comment:
this kinda smells like it needs a little struct of its own? i.e., something to capture the three things?

Reply:
your wish...